File: eSalon/eSalon.Model/Rezervacija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eSalon.Model
{
    public class Rezervacija
    {
        public int RezervacijaId { get; set; }

        public int KorisnikId { get; set; }

        public int FrizerId { get; set; }

        public string Sifra { get; set; } = null!;

        public DateTime DatumRezervacije { get; set; }

        public TimeSpan VrijemePocetka { get; set; }

        public TimeSpan? VrijemeKraja { get; set; }
        public bool TerminZatvoren { get; set; }

        public decimal UkupnaCijena { get; set; }

        public int? UkupnoTrajanje { get; set; }

        public int? UkupanBrojUsluga { get; set; }

        public string? StateMachine { get; set; }

        public int NacinPlacanjaId { get; set; }

        public int? AktiviranaPromocijaId { get; set; }

        public string? FrizerImePrezime { get; set; }
        public string? KlijentImePrezime { get; set; }

        public string? NacinPlacanjaNaziv { get; set; }
        public  List<StavkeRezervacije> StavkeRezervacijes { get; set; } = new List<StavkeRezervacije>(); 
    }
}

File: eSalon/UI/esalon_mobile/lib/models/rezervacija.dart
import 'package:esalon_mobile/models/stavke_rezervacije.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int? rezervacijaId;
  int? korisnikId;
  int? frizerId;
  String? sifra;
  DateTime? datumRezervacije;
  String? vrijemePocetka; 
  String? vrijemeKraja; 
  bool? terminZatvoren;
  double? ukupnaCijena;
  int? ukupnoTrajanje;
  int? ukupanBrojUsluga;
  String? stateMachine;
  int? nacinPlacanjaId;
  int? aktiviranaPromocijaId;
  String? aktiviranaPromocijaNaziv;
  String? frizerImePrezime;
  String? klijentImePrezime;
  String? nacinPlacanjaNaziv;
  List<StavkeRezervacije>? stavkeRezervacijes;

  Rezervacija({
    this.rezervacijaId,
    this.korisnikId,
    this.frizerId,
    this.sifra,
    this.datumRezervacije,
    this.vrijemePocetka,
    this.vrijemeKraja,
    this.terminZatvoren,
    this.ukupnaCijena,
    this.ukupnoTrajanje,
    this.ukupanBrojUsluga,
    this.stateMachine,
    this.nacinPlacanjaId,
    this.aktiviranaPromocijaId,
    this.aktiviranaPromocijaNaziv,
    this.frizerImePrezime,
    this.klijentImePrezime,
    this.nacinPlacanjaNaziv,
    this.stavkeRezervacijes,
  });

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}